ast/desugar/Desugar.cc

```diff
@@ -404,6 +404,41 @@ TreePtr desugarMlhs(DesugarContext dctx, core::LocOffsets loc, parser::Mlhs *lhs
     return MK::InsSeq(loc, std::move(stats), MK::Local(loc, tempRhs));
 }
 
+// Map all MatchVars used in `pattern` to local variables initialized from magic calls
+void desugarPatternMatchingVars(InsSeq::STATS_store &vars, DesugarContext dctx, unique_ptr<parser::Node> &node) {
+    if (auto var = parser::cast_node<parser::MatchVar>(node.get())) {
+        auto loc = var->loc;
+        auto recv = MK::Constant(loc, core::Symbols::Magic());
+        auto val = MK::Send0(loc, std::move(recv), core::Names::patternMatchVar());
```

Comment:
This intrinsic just returns `T.untyped`, right? Is there any point to it? It takes no arguments so it doesn't even keep track of enough original state for a later part of the pipeline to be able to learn the type.

With that in mind, can we make this more obvious that it's doing something wrong? Like `ast::MK::Unimplemented(loc)`? That helper makes an untyped value just the same.

Reply:
Replaced by `RaiseUnimplemented`.